core-web/libs/sdk/react/src/lib/components/Container/Container.tsx

```diff
@@ -100,6 +100,10 @@ export function Container({ containerRef }: ContainerProps) {
     }
 
     const renderContentlets = updatedContentlets.map((contentlet) => {
+        if (contentlet.identifier === 'TEMP_EMPTY_CONTENTLET' && !isInsideEditor) {
+            return; // If is the temp empty contenlet and is not inside the editor, don't render anything
+        }
```

Comment:
Is it ok for React to render a map that some entries are `undefined`? Are we sure about this? Because this array will be like `[Component, Compoment, undefined, Component]`.

Reply:
There don't seem to be any problems rendering a null, but I'm going to filter out nulls to avoid future errors

I'll add a filter after the map because I assume there won't be a significant amount of contentlets to cause memory leaks.

Or if there is a possibility of this (a big amount of contentlets)?

Reply:
I think we can prevent to have `undefined` in the array to render by just do the condition around line 55 where we inject the empty contentlet we should just not inject empty contentlets if we are in production.

Also:

<img width="426" alt="image" src="https://github.com/dotCMS/core/assets/751424/6d94703c-5ba9-40fc-a662-1a15fec4a1b2">

Reply:
Good approach, done!

Reply:
React will proabably get confused by a null/undefined entry, mostly because if it doesn't find a key for that position of the array it will use the index as a way to identify the node in the DOM and if for some reason those JSX nodes change it will break the rendering order or the whole rendering process, because that undefined/null doesn't have any reference for react to track changes, this can happen with a JSX node without key prop, this is not about the null or undefined is mostly about how react works with array of JSX nodes

Reply:
It's always recommended to pass a key to all JSX nodes that are rendered using JS maps, because React is really dumb in handling nodes without an static reference to them